source/util/timer.cpp

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) 2018 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <sys/resource.h>
+#include <sys/time.h>
+#include <iomanip>
+#include <iostream>
+
+#include "util/timer.h"
+
+namespace spvutils {
+
+#if defined(SPIRV_TIMER_ENABLED)
+
+namespace {
+inline double TimeDifference(const timeval& before, const timeval& after) {
+  return static_cast<double>(after.tv_sec - before.tv_sec) +
+         static_cast<double>((after.tv_usec - before.tv_usec) / 10000) * .01;
```

Comment:
Why split the arithmetic with tv_usec?  It's suseconds_t and that is an integral type.

I would cast to double and then divide by a million.

Reply:
I wanted to show two digits after decimal point. I forgot that I already did it by adopting std::ios_base::precision. I updated it.